lib/mongo.go

```diff
@@ -114,16 +122,23 @@ func getIdFromResult(result map[string]interface{}) (string, error) {
 func getPropertiesMapFromResult(result map[string]interface{}, c *Collection) map[string]interface{} {
 	properties := make(map[string]interface{})
 	for fieldName, field := range c.Fields {
-		value := getForNestedKey(result, field.Source)
+		value := getForNestedKey(result, fieldName)
 
-		// The destination name (e.g. name of the collection in the warehouse) can be set by the user,
-		// otherwise it just defaults to the collection name in Mongo.
+		// The field name (e.g. name of the field in the warehouse) can be set by the user,
+		// otherwise it just defaults to the field name in Mongo.
 		destinationName := fieldName
-		if field.DestinationName != "" {
+		if field != nil && field.DestinationName != "" {
 			destinationName = field.DestinationName
 		}
 
-		if value != nil && value != bson.Undefined {
+		if _, ok := value.([]interface{}); ok {
```

Comment:
Can you drop a short comment here about how we flatten arrays into a string, and then also ignore / do not upload fields that do not exist for a doc